mlflow/spark.py

```diff
@@ -451,9 +452,13 @@ def _save_model_metadata(
 def _validate_model(spark_model):
     from pyspark.ml.util import MLReadable, MLWritable
     from pyspark.ml import Model as PySparkModel
+    from pyspark.ml import Transformer as PySparkTransformer
 
     if (
-        not isinstance(spark_model, PySparkModel)
+        (
+            not isinstance(spark_model, PySparkModel)
+            and not isinstance(spark_model, PySparkTransformer)
```

Comment:
I'm a bit confused and my spark knowledge is very rusty at this point, but didn't Model inherit from Transformer? Can't we just move the check up a level in the type tree instead of checking either type?

Reply:
We could, but we might run into issues in the event of a breaking change to the inheritance hierarchy. However unlikely that may be, I like the robustness of the current approach to that case.